### PR TITLE
B-18412 Remove edit & review request buttons/links on MTO for DDDSIT service item

### PR DESCRIPTION
**Pull request by danieljordan-caci**
## Summary

As we transition to making the SIT final destination address to be what is at the shipment level, we need to ensure that office users can't update the final destination address at the service item level. Currently this functionality exists and this story does the following:

- remove any functionality that allows any office user to update the final destination address of a destination SIT service item
- take care of any failing tests when deleting code
- handle tech debt and delete components that are no longer in use

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access MM office
2. Login as a TOO
3. You will find that all SIT destination address flags, modals, and components are no longer accessible and removed from the codebase

### CODE REVIEW

**Comment by danieljordan-caci** [comment]
> # Test coverage on server failed.
> **Last run: Thu 28 Dec 2023 09:44:20 PM UTC** Refer to [this report](https://output.circle-artifacts.com/output/job/3d672075-a6e4-4334-8165-27b845fdec57/artifacts/0/test-results/gotest/app/go-coverage.html) to see coverage details for the server app. Documentation for test coverage can be found [at this link](https://transcom.github.io/mymove-docs/docs/about/development/test-coverage-reports).

LOL ofcourse I'd fail server test coverage on a PR that doesn't even touch backend code

**Comment by deandreJones** [comment]
> > # Test coverage on server failed.
> > **Last run: Thu 28 Dec 2023 09:44:20 PM UTC** Refer to [this report](https://output.circle-artifacts.com/output/job/3d672075-a6e4-4334-8165-27b845fdec57/artifacts/0/test-results/gotest/app/go-coverage.html) to see coverage details for the server app. Documentation for test coverage can be found [at this link](https://transcom.github.io/mymove-docs/docs/about/development/test-coverage-reports).
> 
> LOL ofcourse I'd fail server test coverage on a PR that doesn't even touch backend code

i dont think you actually failed test coverage- but some other issue- looks liek a circle fail
reading url

**Comment by deandreJones** [comment]
also looks like you need to modify some e2e tests

File: src/components/Office/ServiceItemsTable/ServiceItemsTable.jsx [hunk 1]
```diff
@@ -255,40 +217,26 @@ const ServiceItemsTable = ({
                     </span>{' '}
                     Reject
                   </Button>
-                  {ALLOWED_SIT_ADDRESS_UPDATE_SI_CODES.includes(code) && (
+                  {ALLOWED_SIT_UPDATE_SI_CODES.includes(code) && (
                     <div>
-                      {sitAddressUpdates && hasSITAddressUpdate(sitAddressUpdates) ? (
-                        <Button
-                          type="button"
-                          data-testid="reviewRequestTextButton"
```
Comment (loganwc) [inline src/components/Office/ServiceItemsTable/ServiceItemsTable.jsx:263]: looks like this is being referenced in laywright/tests/office/txo/sitUpdates.spec.js:244:78

Comment (danieljordan-caci) [reply src/components/Office/ServiceItemsTable/ServiceItemsTable.jsx:263]: Thanks for the spot! New push just sent.

**Comment by danieljordan-caci** [comment]
For some reason server test coverage is failing. If @deandreJones or @cameroncaci approve, this can be bypassed when pushed to staging since it is not a mandatory passing check with that circleCI build.

I don't plan on adding any server tests to this PR since it did not touch any server code.

**Comment by deandreJones** [comment]
> For some reason server test coverage is failing. If @deandreJones or @cameroncaci approve, this can be bypassed when pushed to staging since it is not a mandatory passing check with that circleCI build.
> 
> I don't plan on adding any server tests to this PR since it did not touch any server code.

image

**Comment by danieljordan-caci** [comment]
> > For some reason server test coverage is failing. If @deandreJones or @cameroncaci approve, this can be bypassed when pushed to staging since it is not a mandatory passing check with that circleCI build.
> > I don't plan on adding any server tests to this PR since it did not touch any server code.
> 
> image

but WHAT DOES IT MEAN DRE?

**Comment by deandreJones** [comment]
> > > For some reason server test coverage is failing. If @deandreJones or @cameroncaci approve, this can be bypassed when pushed to staging since it is not a mandatory passing check with that circleCI build.
> > > I don't plan on adding any server tests to this PR since it did not touch any server code.
> > 
> > 
> > image
> 
> but WHAT DOES IT MEAN DRE?

im not quite sure- other things have gotten past this step without the error - so something is odd with this one, i dont see anythign obvious with your changes that would lead to this error, but im looking

**Comment by deandreJones** [comment]
i think this is fine to merge- its failing trying to delete the server test coverage comment that the bot added when it orignally failed, but that comment is no longer showing so its failing to delete it...

**Review by deandreJones — APPROVED** [review]

**Comment by danieljordan-caci** [comment]
@deandreJones can you check out Happo?

**Review by cameroncaci — APPROVED** [review]
A shame so much code has to go, but it aligns with the ticket description and AC.

Tested successfully on this end, code looks good.

**Pull request merged**